src/Fantomas/CodePrinter.fs

```diff
@@ -1698,7 +1698,18 @@ and genExpr astContext synExpr ctx =
                 (match e with
                  | SynExpr.IfThenElse _
                  | SynExpr.Match _ when (ctx.Config.IndentSize <= operatorText.Length) ->
-                     autoParenthesisIfExpressionExceedsPageWidth (genExpr astContext e)
+                     (fun ctx ->
+                         let ctxAfterMatch = genExpr astContext e ctx
+
+                         let barOnLastLine =
```

Comment:
@jindraivanek I'm not yet happy the way I check if the last clause was multiline or not.
This would check the last line `| Shared.Barry -> "barry"` and continue if the `| ` matches.
Feels like that wouldn't cover everything. Do you see any alternatives?